File: Creational/AbstractFactory/PHP/factories/MacFactory.php
<?php

require_once(__DIR__.'/../interfaces/IGUIFactory.php');
require_once(__DIR__.'/../interfaces/IButton.php');
require_once(__DIR__.'/../interfaces/ICheckbox.php');
require_once(__DIR__.'/../products/MacButton.php');
require_once(__DIR__.'/../products/MacCheckbox.php');

/**
 * Concrette factory.
 */

class MacFactory implements IGUIFactory
{
	/**
	 * @override
	 */
	public function createButton(): IButton
	{
		return new MacButton();
	}

	/**
	 * @override
	 */
	public function createCheckbox(): ICheckbox
	{
		return new MacCheckbox();
	}
}

File: Creational/AbstractFactory/C++/factories/MacFactory.cpp
#include "MacFactory.h"
#include "./../interfaces/IButton.h"
#include "./../interfaces/ICheckbox.h"
#include "./../products/MacButton.h"
#include "./../products/MacCheckbox.h"

/**
 * Concrette factory (realisation).
 */

IButton* MacFactory::createButton()
{
	return new MacButton();
}

ICheckbox* MacFactory::createCheckbox()
{
	return new MacCheckbox();
}
